indico/modules/events/registration/util.py

```diff
@@ -1030,3 +1033,39 @@ def process_registration_picture(source, *, thumbnail=False):
     picture.save(image_bytes, 'JPEG')
     image_bytes.seek(0)
     return image_bytes
+
+
+def create_registration_check_type(data, event=None):
+    check_type = RegistrationCheckType(event=event, is_system_defined=not event)
+    check_type.populate_from_dict(data)
+    db.session.add(check_type)
+    db.session.flush()
+    logger.info('Registration check type "%s" created by %s', check_type, session.user)
+    return check_type
+
+
+def update_registration_check_type(check_type, data):
+    check_type.populate_from_dict(data)
+    db.session.flush()
+    logger.info('Registration check type "%s" updated by %s', check_type, session.user)
+
+
+def delete_registration_check_type(check_type):
+    db.session.delete(check_type)
+    db.session.flush()
+    logger.info('Registration check type "%s" deleted by %s', check_type, session.user)
+
+
+def create_registration_check(registration, check_type=None, is_check_out=False, checked_by_user=None):
+    if not check_type:
+        check_type = registration.event.default_check_type
+    result = registration.can_perform_check(check_type=check_type, is_check_out=is_check_out)
+    if not result['can_check']:
+        raise ValueError(result['reason'])
+    check = RegistrationCheck(registration=registration, check_type=check_type, is_check_out=is_check_out,
+                              checked_by_user=checked_by_user)
+    db.session.add(check)
+    db.session.commit()
```

Comment:
Why not rely on the RH to commit? If you just need it to have an id for some reason, simply flush here.

Reply:
Right!